tests/driver_lsm6dsl/main.c

```diff
@@ -12,28 +12,590 @@
+        if (lsm6dsl_acc_enable_alert(&dev, LSM6DSL_INT_SINGLE_TAP,
+            atoi(argv[2]), alert_cb, "DoubleTAP") != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+        puts(" To trig the alert, single/double tap on your device");
+    }
+    else {
+        _usage_enable();
+        return 1;
+    }
+
+    return 0;
+}
+
+static void _usage_disable(void)
+{
+    printf("Usage: disable <wkup_int | sleep_int | 6d4d_int | ff_int | " \
+            "tap1_int | tap2_int> <pin> \n");
+    puts("");
+}
+
+static int _lsm6dsl_disable_handler(int argc, char **argv)
+{
+    if (argc < 3) {
+        _usage_disable();
+        return 1;
+    }
+    if (strcmp(argv[1], "wkup_int") == 0) {
+        if (lsm6dsl_acc_disable_alert(&dev, LSM6DSL_INT_WAKE_UP, atoi(argv[2]))
+            != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+    }
+    else if (strcmp(argv[1], "sleep_int") == 0) {
+        if (lsm6dsl_acc_disable_alert(&dev, LSM6DSL_INT_SLEEP, atoi(argv[2]))
+            != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+    }
+    else if (strcmp(argv[1], "ff_int") == 0) {
+        if (lsm6dsl_acc_disable_alert(&dev, LSM6DSL_INT_FREE_FALL,
+            atoi(argv[2]))
+            != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+    }
+    else if (strcmp(argv[1], "6d4d_int") == 0) {
+        if (lsm6dsl_acc_disable_alert(&dev, LSM6DSL_INT_6D_4D, atoi(argv[2]))
+                != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+    }
+    else if (strcmp(argv[1], "tap1_int") == 0) {
+        if (lsm6dsl_acc_disable_alert(&dev, LSM6DSL_INT_SINGLE_TAP,
+            atoi(argv[2]))
+            != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+    }
+    else if (strcmp(argv[1], "tap2_int") == 0) {
+        if (lsm6dsl_acc_disable_alert(&dev, LSM6DSL_INT_SINGLE_TAP,
+            atoi(argv[2]))
+            != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+        puts("[SUCCESS]\n");
+    }
+    else {
+        _usage_enable();
+        return 1;
+    }
+
+    return 0;
+}
+
+static void _usage_get_reason(void)
+{
+    printf("Usage: get_reason <wkup | sleep | 6d4d | ff | tap>");
+    puts("");
+}
+
+static int _lsm6dsl_get_reason_handler(int argc, char **argv)
+{
+    if (argc < 2) {
+        _usage_get_reason();
+        return 1;
+    }
+    if (strcmp(argv[1], "wkup") == 0 || strcmp(argv[1], "sleep") == 0) {
+        lsm6dsl_wkup_src_t wkup;
+        if (lsm6dsl_acc_get_wkup_src_alert(&dev, &wkup, false) != LSM6DSL_OK) {
+            puts("[ERROR]");
+            return 1;
+        }
+
+        printf(" Wake up reason :\n");
```

Comment:
I tried wkup event on the b-l475e-iot01a (following the readme) and it did trigger as expected. But afterward the reason values are all 0.